src/screens/FilterModal/Header.js

```diff
@@ -0,0 +1,57 @@
+// @flow
+import React from "react";
+import { StyleSheet, View } from "react-native";
+import SafeAreaView from "react-native-safe-area-view";
+import ActionButton from "../../components/ActionButton";
+import Text from "../../components/Text";
+import ContentPadding from "../../components/ContentPadding";
+import { lightNavyBlueColor, whiteColor } from "../../constants/colors";
+import text from "../../constants/text";
+
+type Props = {
+  onClearPress: () => void,
+  onCancelPress: () => void,
+  showClear: boolean
+};
+
+const Header = ({ onClearPress, onCancelPress, showClear }: Props) => (
+  <SafeAreaView style={styles.container} forceInset={{ top: "always" }}>
+    <ContentPadding style={styles.content}>
+      <ActionButton label={text.cancel} onPress={onCancelPress} />
+      <View style={styles.titleWrapper}>
+        <Text
+          type="h2"
+          style={styles.titleText}
+          accessibilityTraits={["header"]}
+        >
+          {text.filterEvents}
+        </Text>
+      </View>
+      <ActionButton
+        key={showClear ? 1 : 0}
```

Comment:
What does the `key` do?

Reply:
Strangely. I needed the key to make it re-render. But I've refactored the components since then so I'll double check it's needed.

Reply:
Yea I've checked this again and strangely the component does not re-render when the style prop changes. Changing the key forces the renderer to take notice of the change. I'll add a comment in the code to make it clearer.